lib/MergeDuplicateProperties.js

```diff
@@ -0,0 +1,383 @@
+'use strict';
+var Cesium = require('cesium');
+var clone = require('clone');
+var bufferEqual = require('buffer-equal');
+var deepEqual = require('deep-equal');
+
+var byteLengthForComponentType = require('./byteLengthForComponentType');
+var getAccessorByteStride = require('./getAccessorByteStride');
+var numberOfComponentsForType = require('./numberOfComponentsForType');
+
+var defined = Cesium.defined;
+
+module.exports = MergeDuplicateProperties;
+
+/**
+ * Contains stages to merge duplicate glTF elements.
+ * A duplicate element is one that could be substituted by another existing element with no behavior change.
+ * Duplicate references are only re-assigned, the redundant elements are not deleted.
+ *
+ * @constructor
+ */
+function MergeDuplicateProperties() {}
+
+/**
+ * Merges all duplicate elements in the glTF asset in top-down order so merged references resolve down the hierarchy.
+ *
+ * @param {Object} gltf A javascript object containing a glTF asset.
+ * @returns {Object} The glTF asset with merged duplicate elements.
+ */
+MergeDuplicateProperties.mergeAllDuplicates = function(gltf) {
+    MergeDuplicateProperties.mergeDuplicateAccessors(gltf);
```

Comment:
The names are a bit redundant (as I imagine they are for removing unused).  It's a bit awkward because the class is a verb, not a noun.  Maybe:
```
MergeDuplicateProperties.mergeDuplicateAccessors -> MergeDuplicateProperties.mergeAccessors
MergeDuplicateProperties.mergeDuplicateAccessors -> DuplicateEliminator.mergeAccessors
MergeDuplicateProperties.mergeDuplicateAccessors -> DuplicateOptimizer.mergeAccessors
```
Consider a similar rename for removing unused; breaking the API is no problem at this point.